### PR TITLE
Contents addition to server developer's guide

### PR DESCRIPTION
**Pull request by shimono**

### CODE REVIEW

File: en/software-developer/README.md [hunk 1]
```diff
@@ -1,46 +1,67 @@
...
+Personium project thinks it is important to distribute our product under Apache 2.0 license so that the users can freely and safely use it.
+So we have a policy that our product should be consisted only of libraries compatible with the same license.
+To be more specific, we do not use the libraries with a license that include redistribution limitation terms.
 
-[personium-engine](https://github.com/personium/personium-engine)
+## Testing for quality
 
-[personium-lib-common](https://github.com/personium/personium-lib-common)
+* Personium server software assures its quality with [Automated Tests](./personium_tests.md) 
+* When you make a Pull Resuest, please issue it after running relevant part of the automated tests, and making sure there are no errors.
```
Comment (dixonsiu) [inline en/software-developer/README.md]: Spelling mistakes. Should be "Pull Request".

File: en/software-developer/personium_tests.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+|:--|:--|
+|Unit Test|Checks the behavior of each class / method.|
+|Integration Test|Checks the Web API behavior.|
+
+
+## Unit Test
+
+Checks the behavior of each class / method. 
+Keep the tests as simple as we can by using mocks.
+Do not communicate with external programs such as ElasticSearch, ActiveMQ.
```
Comment (dixonsiu) [inline en/software-developer/personium_tests.md]: Suggestion: Do not write Unit tests that communicate with external programs such as Personium Engine, MemCached, ElasticSearch or ActiveMQ.

File: ja/software-developer/personium_tests.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+### パッケージ/クラス/メソッドの命名規約
+
+|対象|命名規約|備考|
+|:--|:--|:--|
+|パッケージ|対象クラスと同じ|io.personium.core.*となる|
+|クラス名|{テスト対象クラス名}Test.java| |
+|メソッド名|{テスト対象メソッド名}_{条件}_{結果}| |
+
```
Comment (dixonsiu) [inline ja/software-developer/personium_tests.md]: Should add an example to avoid confusion.

File: en/software-developer/personium_tests.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+### Package / Class / Method naming conventions
+
+|Target|Convention|Note|
+|:--|:--|:--|
+|Package|same as test target class|io.personium.core.*|
+|Class Name|{target class name}Test.java| |
+|Method Name|{target Method name}_{Conditions}_{Result}| |
```
Comment (tochi-y) [inline en/software-developer/personium_tests.md]: Needs to escape _ in markdown.

```suggestion
|Method Name|{target Method name}\_{Conditions}\_{Result}| |
```

File: en/software-developer/personium_tests.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+- Use English language for all of above.
+
+## Integration Test
+
+Test programs under io.personium.test.* packages.
+They check the behavior of each Web API, communicating with external programs such as ElasticSearch, ActiveMQ.
+
+### Preparation
+
+Integration test can run correctly only after
```
Comment (tochi-y) [inline en/software-developer/personium_tests.md]: Note that new lines are ignored in markdown. If you display a new line, add 2 spaces into end of a line.

File: en/software-developer/personium_tests.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+## Integration Test
+
+Test programs under io.personium.test.* packages.
+They check the behavior of each Web API, communicating with external programs such as ElasticSearch, ActiveMQ.
+
+### Preparation
+
+Integration test can run correctly only after 
+correct test data are prepared.
+By running Setup#reset(), test data will be prepared.
```
Comment (tochi-y) [inline en/software-developer/personium_tests.md]:
```suggestion
By running Setup#reset() and Setup#resetEventLog(), test data will be prepared.
```

File: en/software-developer/README.md (hunk 1 above)
Comment (dixonsiu) [reply en/software-developer/README.md]:
```suggestion
* When you make a Pull Request, please issue it after running relevant part of the automated tests, and making sure there are no errors.
```

File: ja/software-developer/personium_tests.md (hunk 3 above)
Comment (shimono) [reply ja/software-developer/personium_tests.md]: Yes I would like to add examples later

**Pull request merged**